functions/activate-batch-background.js

```diff
@@ -65,14 +81,18 @@ async function handle(data, db, contract) {
     console.log('### Address shuffle round 3 ###')
     shuffledAddresses = shuffle(shuffledAddresses)
 
-    // Slice list for extra addresses
-    shuffledAddresses = shuffledAddresses.slice(0, prizes.length)
-
-    // Assign prizes based on addresses list shuffle
     let prizeAssignment = {}
     for(let i = 0;i < shuffledAddresses.length;i++) {
-      const val = prizeAssignment[shuffledAddresses[i]]
-      prizeAssignment[shuffledAddresses[i]] = [...(val ? val : []), prizes[i].name]
+      if (i < prizes.length) {
+        // Assign prizes based on addresses list shuffle. These are winners
+        const val = prizeAssignment[shuffledAddresses[i]]
+        prizeAssignment[shuffledAddresses[i]] = [...(val ? val : []), prizes[i].name]
+        // Increase badluck count for winners, in proportion to their number of nanopasses
```

Comment:
This section needs to be moved above the `if (i < prizes.length)` but still inside the loop, so it can initialise the non-winners with an empty array